pkg/cosign/verify.go

```diff
@@ -677,6 +680,11 @@ func verifyInternal(ctx context.Context, sig oci.Signature, h v1.Hash,
 		return false, fmt.Errorf("unable to verify bundle: %w", err)
 	}
 
+	// If the --offline flag was specified, fail here
+	if !bundleVerified && co.Offline {
```

Comment:
i think the line above needs to be changed, or else the bundle verification error output would not make sense
```
	if err != nil && co.RekorClient == nil {
```

like if the bundle is not verified and we have a RekorClient set (which can still happen with offline mode so far), then the bundle verification error doesn't get propogated.

Reply:
Arguably you shouldn't have  RekorClient set in offline, so the two checks might merge if you enforce that.

Reply:
It seems like currently, setting `rekor-url` to `""` is equivalent to an offline flag. Do we want to remove that assumption? I think it should be an error to set `rekor-url=""`, and that the source of truth for whether we do offline or not will be this new `offline` flag

Reply:
yah since rekor-url has a default value RekorClient is pretty much always set (maybe people set rekor-url="" but i haven't really seen evidence of that). 

> the source of truth for whether we do offline or not will be this new offline flag
+1

i can update the error so that it propagates.

Reply:
@znewman01 i believe it's updated here

Reply:
maybe a GH UI hiccup, weird. thx!